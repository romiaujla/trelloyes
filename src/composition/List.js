import React from 'react';
import Card from './Card.js';

function AllCards(props){
    const allCards = props.cards.map(card => <Card title={card.title} content={card.content} />);
    return (
        <ul className="card-list">
            {allCards}
        </ul>
    );
}

function List(props){
    return (
        <section className="cards-list-section">
            <Header className="cards-list-header">
                {props.header}
            </Header>
            <AllCards cards={props.cards} />
        </section>
    );
}

export default List;