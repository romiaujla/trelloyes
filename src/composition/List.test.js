import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List Component', ()=>{


    const cardData = [
        {
            id: 0,
            title: 'First Card',
            content: 'lorem ipsum'
        },
        {
            id: 1,
            title: 'Second Card',
            content: 'lorem ipsum'
        }
    ];

    it('renders list without crashing', ()=>{
        const div = document.createElement('div');
        ReactDOM.render(<List header='First List' cards={cardData} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders list correctly when props are passed', ()=>{
        const tree = renderer
            .create(<List header="First List" cards={cardData} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});