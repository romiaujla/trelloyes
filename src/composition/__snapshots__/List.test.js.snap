// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders list correctly when props are passed 1`] = `
<section
  className="cards-list-section"
>
  <header
    className="cards-list-header"
  >
    <h2>
      First List
    </h2>
  </header>
  <ul
    className="card-list"
  >
    <li
      className="card"
    >
      <div
        className="flex-div"
      >
        <h3>
          First Card
        </h3>
        <button
          className="card-delete-button"
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </li>
    <li
      className="card"
    >
      <div
        className="flex-div"
      >
        <h3>
          Second Card
        </h3>
        <button
          className="card-delete-button"
          type="button"
        >
          delete
        </button>
      </div>
      <p>
        lorem ipsum
      </p>
    </li>
  </ul>
  <div
    className="add-random-card"
  >
    <button
      className="add-random-card-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
